apps/src/randomPointGenerator.cpp

```diff
@@ -1,44 +1,40 @@
+#include <fstream>
 #include <iostream>
 #include <random>
-#include <fstream>
 
-// returns random int on interval [a,b]
-int RandomInt(int a, int b)
-{
-    std::random_device genDevice;
-    std::uniform_int_distribution<int> genDist(a, b);
-    return genDist(genDevice);
-}
+#include <boost/filesystem.hpp>
 
-// argv == et_PointGenerator ymin ymax xmin xmax
+#include "envitools/CSVIO.hpp"
+#include "envitools/EnviUtils.hpp"
+
+// argv == et_PointGenerator xmin, ymin, xmax, ymax, output path
 int main(int argc, char** argv) {
-	if (argc < 5) {
-		std::cout << " Usage: " << argv[0];
-		std::cout << " ymin ymax xmin xmax" << std::endl;
-		return EXIT_FAILURE;
-	}
-	
-	int num_points = 500;
-	int xmin = [3];
-	int ymin = [1];
-	int xmax = [4];
-	int ymax = argv[2];
-
-	std::ofstream myfile;
-	myfile.open("papyrus_points.txt");
-
-	std::cout << "X,Y" << std::endl;
-	myfile << "X,Y" << std::endl;
-	
-	for (int i = 0; i < num_points; i++)
-	{
-		int x = RandomInt(xmin, xmax);
-		int y = RandomInt(ymin, ymax);
-
-		std::cout << x << "," << y << std::endl;
-		myfile << x << "," << y << std::endl;
-	}
-	
-	myfile.close();
-	return 0;
+    if (argc < 6) {
+        std::cout << " Usage: " << argv[0];
+        std::cout << " [xmin] [ymin] [xmax] [ymax] [output path]" << std::endl;
+        return EXIT_FAILURE;
+    }
+
+    int num_points = 500;
+    int xmin = std::stoi(argv[1]);
+    int ymin = std::stoi(argv[2]);
+    int xmax = std::stoi(argv[3]);
+    int ymax = std::stoi(argv[4]);
+
+    std::cout << "X,Y" << std::endl;
+    std::vector<cv::Vec2i> vec;
+    for (int i = 0; i < num_points; i++) {
+        int x = envitools::RandomInt(xmin, xmax);
+        int y = envitools::RandomInt(ymin, ymax);
+        vec.emplace_back(x, y);
+        std::cout << x << "," << y << std::endl;
+    }
+
+    boost::filesystem::path path = argv[5];
+
+    if (boost::filesystem::exists(path)) {
```

Comment:
`!exists`. As written, we never write the CSV _unless_ we're overwriting a file, which is not what we want.